tests/components/test_snips.py

```diff
@@ -1,45 +1,227 @@
+                    }
+                }
+            }
+        })
+
+        service_calls = mock_service(self.hass, 'test', 'snips')
+        payload = """
         {
-            "slotName": "light_color",
-            "value": {
-                "kind": "Custom",
-                "value": "green"
+            "input": "zodiac forecast for virgo",
+            "sessionId": "abcdef1234567890",
+            "intent": {
+                "intentName": "CallServiceIntent"
+            },
+            "slots": [
+                {
+                    "slotName": "ZodiacSign",
+                    "rawValue": "virgo",
+                    "value": {
+                        "kind": "custom",
+                        "value": "virgo"
+                    }
+                }
+            ]
+        }
+        """
+        fire_mqtt_message(self.hass, 'hermes/intent/CallServiceIntent',
+                          payload)
+        self.hass.block_till_done()
+        self.assertEqual(1, len(service_calls))
+        self.assertEqual('virgo', service_calls[0].data['zodiac_sign'])
+        self.assertEqual('switch.test', service_calls[0].data['entity_id'][0])
+
+    def test_intent_with_snips_duration(self):
+        """Test intent with snips duration."""
+        mock_component(self.hass, 'mqtt')
+        setup_component(self.hass, snips.DOMAIN, {
+            snips.DOMAIN: {},
+        })
+        setup_component(self.hass, "intent_script", {
+            "intent_script": {
+                "SetTimer": {
+                    "speech": {
+                        "type": "plain",
+                        "text": "Timer set {{ timer_duration }}",
+                    },
+                }
             }
+        })
+        payload = """
+        {
+            "input": "set a timer of five minutes",
+            "intent": {
+                "intentName": "SetTimer"
+            },
+            "slots": [
+                {
+                    "rawValue": "five minutes",
+                    "value": {
+                        "kind": "Duration",
+                        "years": 0,
+                        "quarters": 0,
+                        "months": 0,
+                        "weeks": 0,
+                        "days": 0,
+                        "hours": 0,
+                        "minutes": 5,
+                        "seconds": 0,
+                        "precision": "Exact"
+                    },
+                    "entity": "snips/duration",
+                    "slotName": "timer_duration"
+                }
+            ]
+        }
+        """
+        with self.assertLogs(level='INFO') as test_handle:
+            fire_mqtt_message(self.hass,
+                              'hermes/intent/SetTimer', payload)
+            self.hass.block_till_done()
+            self.assertIn("\"sessionId\": \"default\"",
+                          test_handle.output[1])
+            self.assertIn("\"text\": \"Timer set 300\"}",
```

Comment:
Wouldn't it make more sense to just fire a timer event and see what kind of variables are being sent to `intent.async_handle`

Reply:
Ah, that makes sense. Appreciate the comments, I'll take another run at it. FWIW, I really disliked the approach of basically grepping the logs but couldn't get the mqtt response passed back in, but let me see what I can do.

Reply:
Took a run at it but couldn't figure it out in the time I had.